### PR TITLE
FP-1318: Create Section Pattern via Bootstrap Container Plugin

### PR DESCRIPTION
**Pull request by wesleyboar**
## To Do

- [x] [install `django-cms/djangocms-bootstrap4`](https://github.com/TACC/Core-CMS/pull/431/files#r981674483) at [v3.0.0](https://github.com/django-cms/djangocms-bootstrap4/releases/tag/3.0.0)
- [x] re-test
- ~~fix [django cms error upon adding container](https://github.com/TACC/Core-CMS/assets/62723358/d94cc4ce-c32c-4736-97c4-f8957d504f30)~~
- [x] ping reviewers

## Overview

Quickly create existing TACC Section _pattern_ via already-installed[^2] Django Bootstrap4 Container _plugin_.

 _CMS Admin/Designer has approved (via private review), but implementation needs dev review._

[^2]: The install is given a new requirement (in `requirements.txt`) to fix a minor UI bug.

## Tickets

* [FP-1318](https://jira.tacc.utexas.edu/browse/FP-1318) → [__FP-1478__](https://jira.tacc.utexas.edu/browse/FP-1478)

## Changes

To our default `settings`:
* Add `DJANGOCMS_BOOTSTRAP4_GRID_CONTAINERS` ([docs](https://github.com/django-cms/djangocms-bootstrap4#settings)).
    * to allow user to quickly create light Section or dark Section (as Container or not)
* (Optional) Load `gettext_lazy`.
    * to support translation of choice group labels and values[^1]

To our default requirements:
* Install [fork of djangocms-bootstrap4](https://github.com/tacc-wbomar/djangocms-bootstrap4) that has [approved](https://github.com/django-cms/djangocms-bootstrap4/pull/145) fix.
    * to fix minor UI bug where Container plugin has blank short description for grouped choices[^1]

[^1]: I.e. `<optgroup>` labels and `<option>` values in selection dropdown when `DJANGOCMS_BOOTSTRAP4_GRID_CONTAINERS` has nested tuples.

## Screenshots

https://user-images.githubusercontent.com/62723358/150383016-0a028625-cd86-4400-baca-d7f7e77db301.mov

## Testing

> __local__:  [instructions](https://github.com/TACC/Core-CMS/wiki/Test-CMS-Changes)
> __remote__: https://dev.cep.tacc.utexas.edu/ ([build](https://jenkins01.tacc.utexas.edu/view/WMA%20CEP/job/Core_CMS_Build/360/), [deploy](https://jenkins01.tacc.utexas.edu/job/Core_Portal_Deploy/1042/))

1. On any page, create a Container plugin.
2. Choose some of the eight CONTAINER TYPEs.
3. Confirm an element is added each time with the appropriate `class` attribute.\*

\* To see what the section should **look** like, you may compare to pattern library: [light](https://cep.tacc.utexas.edu/static/ui/components/detail/o-section--style-light.html), [dark](https://cep.tacc.utexas.edu/static/ui/components/detail/o-section--style-dark.html), [muted](https://cep.tacc.utexas.edu/static/ui/components/detail/o-section--style-muted.html).

### Related PRs
- Use Style Plugin Instead & Define Tag Types:
    https://github.com/TACC/Core-CMS/pull/430
- Use Style Plugin Instead & Support Other Classes:
    https://github.com/TACC/Core-CMS/pull/419

### CODE REVIEW

**Comment by rstijerina** [comment]
Nice work on the PR to django-cms! Do we have an idea of when it will be merged?

**Comment by tacc-wbomar** [comment]
Thank you.

We do not have an idea. I'll update here if I learn.

_I've [asked for the status on that PR](https://github.com/django-cms/djangocms-bootstrap4/pull/145#issuecomment-1047260159) and been watching [the DjangoCMS Slack channel](http://www.django-cms.org/slack). Earlier this month, [the team (and that PR's approver) had a heavy PR load]](https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1644749226831949) so I haven't poked anyone. (But I have watched how others poke, so I can be ready to.)_

**Comment by tacc-wbomar** [comment]
@rstijerina I still have not received a reply to my inquiry of when DjangoCMS Bootstrap4's PR would get merged.

[So, I've pinged 'em in their Slack channel, just now.](https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1647017720335469)

**Review by taoteg — APPROVED** [review]
Works as described. Great job!

Screen Shot 2022-03-14 at 12 15 04 PM

**Comment by tacc-wbomar** [comment]
Thanks, @taoteg!

I'm holding off on merging this to see if Bootstrap will merge [my fix to their code](https://github.com/django-cms/djangocms-bootstrap4/pull/145) this spring, so we need not point to and maintain [my fork](https://github.com/tacc-wbomar/djangocms-bootstrap4).

**Comment by wesleyboar** [comment]
Progress on the Django CMS PR: [The approving reviewer pinged for others to double-check.](https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1650206720903099)

**Comment by wesleyboar** [comment]
Pinged again: https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1653582318263189

**Comment by wesleyboar** [comment]
[I may need to test their changes to get it merged.](https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1653677956231229?thread_ts=1653582318.263189&cid=C01J7H87KFC)

**Comment by wesleyboar** [comment]
[2nd reviewer claims he has comments for me.](https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1653675755611629?thread_ts=1653582318.263189&cid=C01J7H87KFC) But [I can't find them](https://django-cmsworkspace.slack.com/archives/C01J7H87KFC/p1654365247775409?thread_ts=1653582318.263189&cid=C01J7H87KFC). I found two issues and noted them in [the PR description](https://github.com/django-cms/djangocms-bootstrap4/pull/145).

_I don't have the bandwidth to tend to it anymore right now._

**Comment by wesleyboar** [comment]
The PR from my fork is merged upstream! I updated this PR to install at the latest commit of the plugin's `main` branch, which is 13 commits (over two years) ahead of 2.0.0. I'll wait a month to see if they cut a new release. if not, I'll retest and ping for review.

**Comment by wesleyboar** [comment]
All right, I want new-to-foss bragging rights for having several commits in [django-cms/djangocms-bootstrap4 v3.0.0](https://github.com/django-cms/djangocms-bootstrap4/releases/tag/3.0.0).

I'll update this PR to v3.0.0 and test when I get a chance.

File: pyproject.toml
Comment (wesleyboar) [inline pyproject.toml:49]: This is installing the `main` branch merge of https://github.com/django-cms/djangocms-bootstrap4/pull/162, instead of a commit within it.

Comment (wesleyboar) [reply pyproject.toml]: No… don't do this now. Maybe it breaks something. This PR shouldn't be to blame.

**Comment by wesleyboar** [comment]
> **Important**
> Still useful. Simpler diff. But [bug since using djangocms-bootstrap4 v3.0.0 + `main`](https://github.com/TACC/Core-CMS/assets/62723358/d94cc4ce-c32c-4736-97c4-f8957d504f30).

**Comment by wesleyboar** [comment]
@rstijerina, yes.

I changed the `djangocms-bootstrap4` version to not be from a fork. Containers cannot be added, but that has been an independent bug ([WP-407](https://tacc-main.atlassian.net/browse/WP-407)).

**Review by rstijerina — APPROVED** [review]

**Pull request merged**
